receiver/oracledbreceiver/scraper.go

```diff
@@ -0,0 +1,394 @@
+	"go.opentelemetry.io/collector/receiver/scrapererror"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/oracledbreceiver/internal/metadata"
+)
+
+const (
+	statsSQL                = "select * from v$sysstat"
+	enqueueDeadlocks        = "enqueue deadlocks"
+	exchangeDeadlocks       = "exchange deadlocks"
+	executeCount            = "execute count"
+	parseCountTotal         = "parse count (total)"
+	parseCountHard          = "parse count (hard)"
+	userCommits             = "user commits"
+	userRollbacks           = "user rollbacks"
+	physicalReads           = "physical reads"
+	sessionLogicalReads     = "session logical reads"
+	cpuTime                 = "CPU used by this session"
+	pgaMemory               = "session pga memory"
+	sessionCountSQL         = "select status, type, count(*) as VALUE FROM v$session GROUP BY status, type"
+	systemResourceLimitsSQL = "select RESOURCE_NAME, CURRENT_UTILIZATION, LIMIT_VALUE, CASE WHEN TRIM(INITIAL_ALLOCATION) LIKE 'UNLIMITED' THEN '-1' ELSE TRIM(INITIAL_ALLOCATION) END as INITIAL_ALLOCATION, CASE WHEN TRIM(LIMIT_VALUE) LIKE 'UNLIMITED' THEN '-1' ELSE TRIM(LIMIT_VALUE) END as LIMIT_VALUE from v$resource_limit"
+	tablespaceUsageSQL      = "select TABLESPACE_NAME, BYTES from DBA_DATA_FILES"
+	tablespaceMaxSpaceSQL   = "select TABLESPACE_NAME, (BLOCK_SIZE*MAX_EXTENTS) AS VALUE FROM DBA_TABLESPACES"
+)
+
+type dbProviderFunc func() (*sql.DB, error)
+
+type clientProviderFunc func(*sql.DB, string, *zap.Logger) dbClient
+
+type scraper struct {
+	statsClient                dbClient
+	tablespaceMaxSpaceClient   dbClient
+	tablespaceUsageClient      dbClient
+	systemResourceLimitsClient dbClient
+	sessionCountClient         dbClient
+	db                         *sql.DB
+	clientProviderFunc         clientProviderFunc
+	metricsBuilder             *metadata.MetricsBuilder
+	dbProviderFunc             dbProviderFunc
+	logger                     *zap.Logger
+	id                         component.ID
+	instanceName               string
+	scrapeCfg                  scraperhelper.ScraperControllerSettings
+	startTime                  pcommon.Timestamp
+	metricsSettings            metadata.MetricsSettings
+}
+
+func newScraper(id component.ID, metricsBuilder *metadata.MetricsBuilder, metricsSettings metadata.MetricsSettings, scrapeCfg scraperhelper.ScraperControllerSettings, logger *zap.Logger, providerFunc dbProviderFunc, clientProviderFunc clientProviderFunc, instanceName string) *scraper {
+	return &scraper{
+		id:                 id,
+		metricsBuilder:     metricsBuilder,
+		metricsSettings:    metricsSettings,
+		scrapeCfg:          scrapeCfg,
+		logger:             logger,
+		dbProviderFunc:     providerFunc,
+		clientProviderFunc: clientProviderFunc,
+		instanceName:       instanceName,
+	}
+}
+
+var _ scraperhelper.Scraper = (*scraper)(nil)
+
+func (s *scraper) ID() component.ID {
```

Comment:
It would be better to use `scraperhelper.NewScraper` to create `scraperhelper.Scraper` instance. No need to implement the interface in that case